group-generators/helpers/data-providers/snapshot/provider.ts

```diff
@@ -216,4 +226,461 @@ export default class SnapshotProvider
+
+      for (const currentChunkAuthor of currentChunkAuthors) {
+        if (!fetchedData[currentChunkAuthor.author]) {
+          fetchedData[currentChunkAuthor.author] = 1;
+        } else {
+          fetchedData[currentChunkAuthor.author] += 1;
+        }
+        created_gt = currentChunkAuthor.created;
+      }
+
+      readline.cursorTo(process.stdout, 0);
+      process.stdout.write(`downloading ... (${downloadNumber})`);
+      downloadNumber += currentChunkAuthors.length;
+    } while (currentChunkAuthors.length > 0);
+
+    return fetchedData;
+  }
+
+  private _querySpaceAuthors(
+    space: string,
+    created_gt = 0,
+    chunkSize = 1000
+  ): Promise<QuerySpaceAuthorsOutput> {
+    return this.query<QuerySpaceAuthorsOutput>(
+      gql`
+        query Authors($space: String!, $created_gt: Int!, $chunkSize: Int!) {
+          proposals(
+            first: $chunkSize
+            where: { space: $space, created_gt: $created_gt }
+            orderBy: "created"
+            orderDirection: asc
+          ) {
+            author
+            created
+          }
+        }
+      `,
+      {
+        space,
+        chunkSize,
+        created_gt,
+      }
+    );
+  }
+
+  public async querySpaceAuthorsCount({
+    space,
+  }: QuerySpaceAuthorsInput): Promise<number> {
+    const authors = await this.querySpaceAuthors({ space });
+    return Object.keys(authors).length;
+  }
+
+  /**
+   * Retrieves space admins from Snapshot.
+   * @param {string} string - space query parameter
+   * @returns {Promise<FetchedData>} - A Promise that resolves to an object containing space admins.
+   */
+  public async querySpaceAdmins({
+    space,
+  }: QuerySpaceAdminsInput): Promise<FetchedData> {
+    const fetchedData: { [address: string]: number } = {};
+    const spaceAdmins = await this._querySpaceAdmins(space);
+    for (const spaceAdmin of spaceAdmins.spaces[0].admins) {
+      fetchedData[spaceAdmin] = 1;
+    }
+    return fetchedData;
+  }
+
+  private _querySpaceAdmins(space: string): Promise<QuerySpaceAdminsOutput> {
+    return this.query<QuerySpaceAdminsOutput>(
+      gql`
+        query Spaces($space: String!) {
+          spaces(where: { id: $space }) {
+            admins
+          }
+        }
+      `,
+      {
+        space,
+      }
+    );
+  }
+
+  public async querySpaceAdminsCount({
+    space,
+  }: QuerySpaceAdminsInput): Promise<number> {
+    const admins = await this.querySpaceAdmins({ space });
+    return Object.keys(admins).length;
+  }
+
+  /**
+   * Retrieves space voters from Snapshot.
+   * This takes a long time because it needs to query all voters in SnapShot
+   *
+   * @param {QuerySpaceVotersAboveXInput} input - Contains parameters for the function:
+   *    @param {string} space - Optional - A string representing the space parameter of the query.
+   *    @param {number} abovex - Optional - A number that defines the minimum number of votes for an author to be included in the result.
```

Comment:
like this no? => A number that defines the minimum number of votes for a **voter** to be included in the result.
\+ there are some empty space/indentation

Reply:
Yes - this was explicit in the GitHub issue so I included it
https://github.com/sismo-core/sismo-hub/issues/1431 

```
{
"name": "number of proposals",
"argName": "TODO",
"type": "TODO",
"example": "3",
"description": "A minimum number of proposals"
},
```

Reply:
Yes so it's "number of votes for a voter" an not:  "number of votes for an author" like what you wrote isn't it?

Reply:
Also I was thinking, wouldn't be better not to create another function for `querySpaceVotersAboveX` & `queryProposalAuthorsAboveX` but have directly another arguments in the `querySpaceVoters` and `querySpaceAuthors` functions that le the user choose if he want a special minimumValue. (but by default set it to 1).
The UX would be way nicer!
This issue has been created when the optional argument wasn't available in the Factory, but that that's not the case anymore, so it makes sens to do this. What do you think?

Reply:
> Yes so it's "number of votes for a voter" an not: "number of votes for an author" like what you wrote isn't it?

Ok got it. I was confused what you were referring to. I'll update the comment